third_party/move/move-core/types/src/value.rs

```diff
@@ -9,19 +9,29 @@ use crate::{
     u256,
 };
 use anyhow::{anyhow, bail, Result as AResult};
+use once_cell::sync::Lazy;
 use serde::{
-    de::{Error as DeError, Unexpected},
-    ser::{SerializeMap, SerializeSeq, SerializeStruct, SerializeTuple},
+    de::{EnumAccess, Error as DeError, VariantAccess},
+    ser::{SerializeMap, SerializeSeq, SerializeStruct, SerializeTuple, SerializeTupleVariant},
     Deserialize, Serialize,
 };
 use std::{
+    collections::{btree_map::Entry, BTreeMap},
     convert::TryInto,
     fmt::{self, Debug},
+    sync::Mutex,
 };
 
+/// The maximal number of enum variants which are supported in values. This must align with
+/// the configuration in the binary format, so the bytecode verifier checks its validness.
+pub const VARIANT_COUNT_MAX: u64 = 127;
+
 /// In the `WithTypes` configuration, a Move struct gets serialized into a Serde struct with this name
 pub const MOVE_STRUCT_NAME: &str = "struct";
 
+/// A Move enum gets serialized into a Serde struct with this name
```

Comment:
QQ will the enum name be part of the serialization result?

Reply:
Nope will not.